From aee40671774ce2a13c6abe3fc0162f0f2e3aff0b Mon Sep 17 00:00:00 2001
From: Richard Guy Briggs <rgb@redhat.com>
Date: Wed, 1 Oct 2014 21:36:01 -0400
Subject: [kernel] audit: fix build error when asm/syscall.h does not exist

Message-id: <d356a186906c4264876e41535d0b261d9b1c8a2e.1412192166.git.rgb@redhat.com>
Patchwork-id: 96865
O-Subject: [RHEL7 PATCH V2 8/9] audit: fix build error when asm/syscall.h does not exist
Bugzilla: 1053921
RH-Acked-by: Eric Paris <eparis@redhat.com>
RH-Acked-by: Paul Moore <pmoore@redhat.com>

From: Eric Paris <eparis@redhat.com>

avr32 does not have an asm/syscall.h file.  We need the
syscall_get_arch() definition from that file for all arch's which
support CONFIG_AUDITSYSCALL.  Obviously avr32 is not one of those
arch's.  Move the include inside the CONFIG_AUDITSYSCALL such that we
only do the include if we need the results.

When the syscall_get_arch() call is moved inside __audit_syscall_entry()
this include can be dropped entirely.  But that is going to require some
assembly changes on x86* in a patch that is not ready for the tree...

Upstream-commit: a9ebe0b
Reported-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Eric Paris <eparis@redhat.com>
Signed-off-by: Richard Guy Briggs <rgb@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/audit.h b/include/linux/audit.h
index 811eece..a3b228e 100644
--- a/include/linux/audit.h
+++ b/include/linux/audit.h
@@ -26,7 +26,6 @@
 #include <linux/sched.h>
 #include <linux/ptrace.h>
 #include <uapi/linux/audit.h>
-#include <asm/syscall.h>
 
 struct audit_sig_info {
  uid_t  uid;
@@ -95,6 +94,8 @@ struct filename;
 extern void audit_log_session_info(struct audit_buffer *ab);
 
 #ifdef CONFIG_AUDITSYSCALL
+#include <asm/syscall.h> /* for syscall_get_arch() */
+
 /* These are defined in auditsc.c */
     /* Public API */
 extern int  audit_alloc(struct task_struct *task);
-- 
1.7.1